httpbakery/agent/agent.go

```diff
@@ -40,6 +41,63 @@ type Agent struct {
 	Username string `json:"username" yaml:"username"`
 }
 
+var ErrNoAuthInfo = errgo.New("no bakery agent info found in environment")
+
+// AuthInfoFromEnvironment returns an AuthInfo derived
+// from environment variables.
+//
+// It recognizes the following variables:
+// BAKERY_PRIVATE_KEY - the private key of the agent, base64 encoded.
+// BAKERY_PUBLIC_KEY - the public key of the agent, base64 encoded.
+// BAKERY_USERS - space-separated (username, URL) pairs of username
+// and identity manager URLs.
+//
+// For example:
+//
+// BAKERY_PUBLIC_KEY='wg/rUMm5cy9O00TaS9pgDIHnFMwhrisI0KDSg4+KkGM='
+// BAKERY_PRIVATE_KEY='S529MFrVUWB/VqcsPABJSAlnY/57fdHi4cL9HdK9ZQw='
+// BAKERY_USERS='https://api.jujucharms.com/identity bob https://api.staging.jujucharms.com/identity'
```

Comment:
This is significantly more complex than I had in mind. I was essentially looking for a fix to https://github.com/juju/charmstore/issues/776. This is not usable that simply. We need to be able to provide API access to jujucharms.com through a mechanism that it "set these two environment variables", "use the tools" type system. The same as people expect from AWS. I'm not interested that we have this crazy bakery setup inside the API that complexity should not be forced onto out user(s).